### DISCUSSION TITLE
Artifacts behavior is not ideal with non-web applications (eg: Python, Jupyter)

### DISCUSSION THREAD

**Discussion by adamrb** [danny-avila/LibreChat#3830] [Feature Requests & Suggestions]
First of all, I love artifacts!  These are great when I want a simple web app for a simple tasks.

I've noticed that the current implementation seems to be geared primarily towards web applications. I recently requested a Python script and a Jupyter notebook, and while the LLM provided the code artifacts as expected, there were a couple of hiccups:

- The code lacked syntax highlighting, which made it a bit harder to read at a glance.
- The system attempted to automatically preview the code, which, unsurprisingly, didn't work for non-web content.

In an ideal world, it'd be great to have a container spun up to execute command-line scripts. But I realize that might be a bit of a stretch for the short term.

For now, I think syntax highlighting and not defaulting to preview for languages where it's not possible would go a long way to help the user experience.

**Comment by danny-avila** [comment]
Hey @adamrb 👋 

I remember you from the early days, it's encouraging to see your handle.

> I've noticed that the current implementation seems to be geared primarily towards web applications. I recently requested a Python script and a Jupyter notebook, and while the LLM provided the code artifacts as expected, there were a couple of hiccups:

It is totally geared towards web apps, same as Claude.ai.

It's actually not intended to run python or anything but JS/HTML/mermaid (which can be translated to HTML)... yet.

> In an ideal world, it'd be great to have a container spun up to execute command-line scripts. But I realize that might be a bit of a stretch for the short term.

I've thought about this problem for a long time. I'm cooking something up for https://github.com/danny-avila/LibreChat/issues/3607

Stay tuned 👀
